### PR TITLE
[ark-lib] test & fix txutils.ReadTxWitness

### PR DESCRIPTION
**Pull request by chris-ricketts**
## Changes:
- Added a bounds check in `ReadTxWitness` to return an error when `witCount` exceeds remaining reader bytes before `make(wire.TxWitness, witCount)`.
- Added `TestReadTxWitness` (table-driven) with:
  - Valid `00` (empty witness)
  - Valid `0100` (one empty item)
  - Valid `01020405` (one item with bytes)
  - Invalid huge-count varint with tiny payload
  - Invalid truncated item (`0102`)

## Summary by CodeRabbit

* **Bug Fixes**
  * Added validation to detect malformed witness data and prevent processing of invalid inputs, improving stability and error reporting.

* **Tests**
  * Added a table-driven test suite covering valid payloads, edge cases, and error scenarios for witness parsing to ensure correctness and prevent regressions.

### CODE REVIEW

File: pkg/ark-lib/txutils/utils_test.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+import (
+	"encoding/hex"
+	"testing"
+
+	"github.com/arkade-os/arkd/pkg/ark-lib/txutils"
+	"github.com/stretchr/testify/require"
+)
+
+func TestReadTxWitness(t *testing.T) {
```
Comment (altafan) [inline pkg/ark-lib/txutils/utils_test.go:11]: also in this pr can you split into valid and invalid?

Comment (chris-ricketts) [reply pkg/ark-lib/txutils/utils_test.go:11]: Fixed in [1a2201a](https://github.com/arkade-os/arkd/pull/974/commits/1a2201ac801b9ff8cb97f592906f5db363d7993f)
